### PR TITLE
Move example images to Artifact Registry

### LINKED ISSUE #2358: Consider moving agones system images from gcr.io to GCP's artifact registry
**Describe the solution you'd like**
GCP's [Artifact Registry](https://cloud.google.com/artifact-registry/) is an alternate way that we can store container images in Google Cloud (we currently store them in Google Container Registry - gcr.io). 

If we put images into the Artifact Registry, we can then take advantage of the recently announced [GKE image streaming for fast application startup and autoscaling](https://cloud.google.com/blog/products/containers-kubernetes/introducing-container-image-streaming-in-gke). This may not be a huge improvement for our linux images (which are pretty small containers) but our windows binaries (the example game server and sdk sidecar) are both pretty large and may see a larger benefit. 

**Describe alternatives you've considered**
* Keep the status quo.
* We can push images to both locations during a migration period.

### PR DESCRIPTION
**Pull request by gongmax**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup

> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:
We want eventually move all of the images generated to Artifact Registry. This PR move all the example images to the Artifact Registry `us-docker.pkg.dev/agones-images/examples`.

**Which issue(s) this PR fixes**:

Closes #2358

**Special notes for your reviewer**:
I have generated and pushed the following images to `us-docker.pkg.dev/agones-images/examples`:
~~~
allocation-endpoint
autoscaler-webhook
cpp-simple
nodejs-simple
rust-simple
unity-simple
xonotic
crd-client
supertuxkart
simple-game-server
~~~

### CODE REVIEW

File: docs/governance/templates/release_issue.md [hunk 1]
```diff
@@ -44,37 +44,35 @@ and copy it into a release issue. Fill in relevant values, found inside {}
...
-   - [ ] Run `make sdk-publish-csharp` to deploy to NuGet. Requires login credentials. (if release candidate: 
-   `make sdk-publish-csharp RELEASE_VERSION={version}-rc`).
-   Will need [NuGet API Key](https://www.nuget.org/account/apikeys) from Agones account.
+    - [ ] Run `make sdk-shell-node` to get interactive shell to publish node package. Requires Google internal process
+      to publish.
+    - [ ] Run `make sdk-publish-csharp` to deploy to NuGet. Requires login credentials. (if release candidate:
+      `make sdk-publish-csharp RELEASE_VERSION={version}-rc`).
+      Will need [NuGet API Key](https://www.nuget.org/account/apikeys) from Agones account.
 - [ ] Run `make do-release`. (if release candidate: `make do-release RELEASE_VERSION={version}-rc`) to create and push the docker images and helm chart.
-- [ ] Run `make shell` and run `gcloud config configurations activate <your development project>` to switch Agones
```
Comment (roberthbailey) [inline docs/governance/templates/release_issue.md:72]: This line was removed - was that on purpose?

Comment (markmandel) [reply docs/governance/templates/release_issue.md:72]: Everything else looks good to me outside of this, so I think we can sneak this into the release? 😄

**Review by markmandel — APPROVED** [review]
LGTM!

**Review by roberthbailey — APPROVED** [review]

File: docs/governance/templates/release_issue.md (hunk 1 above)
Comment (gongmax) [reply docs/governance/templates/release_issue.md:72]: It's removed accidentally. Added it back.

**Pull request merged**